src/main/java/at/hannibal2/skyhanni/features/combat/ArmorStackDisplay.kt

```diff
@@ -3,40 +3,166 @@ package at.hannibal2.skyhanni.features.combat
-import at.hannibal2.skyhanni.utils.RenderUtils.renderString
+import at.hannibal2.skyhanni.utils.RegexUtils.firstMatcher
+import at.hannibal2.skyhanni.utils.RenderUtils.renderStrings
+import at.hannibal2.skyhanni.utils.SimpleTimeMark
+import at.hannibal2.skyhanni.utils.TimeUtils.format
 import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
 import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import kotlin.time.Duration.Companion.milliseconds
 
 @SkyHanniModule
 object ArmorStackDisplay {
-    private val config get() = SkyHanniMod.feature.combat.stackDisplayConfig
-    private var display = ""
+    private val config get() = SkyHanniMod.feature.combat.armorStackDisplay
+    private var stackCount = 0
+    private var stackSymbol = ""
+    private var stackType = ""
+    private var armorPieceCount = 0
+    private var display = listOf<String>()
+    private var stackDecayTimeCurrent = SimpleTimeMark.farPast()
+
+    private val pistonSounds = setOf("tile.piston.out", "tile.piston.in")
 
     /**
      * REGEX-TEST: §66,171/4,422❤  §6§l10ᝐ§r     §a1,295§a❈ Defense     §b525/1,355✎ §3400ʬ
      * REGEX-TEST: §66,171/4,422❤  §65ᝐ     §b-150 Mana (§6Wither Impact§b)     §b1,016/1,355✎ §3400ʬ
      */
-    private val armorStackPattern by RepoPattern.pattern(
+    private val stackPattern by RepoPattern.pattern(
         "combat.armorstack.actionbar",
-        " (?:§6|§6§l)(?<stack>\\d+[ᝐ⁑|҉Ѫ⚶])"
+        " (?:§6|§6§l)(?<stack>\\d+)(?<symbol>[ᝐ⁑|҉Ѫ⚶])",
+    )
+
+    /**
+     * REGEX-TEST: §7Health: §a+205, §7Defense: §a+55, §7Intelligence: §a+125, §7Combat Wisdom: §a+0.75,  §8[§8⚔§8] §8[§8⚔§8], , §bArachno Resistance II, §7Grants §a+3❈ Defense §7against §aspiders§7., §bVeteran I, §7Grants §3+0.75☯ Combat Wisdom§7., , §6Tiered Bonus: Arcane Vision (2/4), §7Gives you the ability to see the runic, §7affinity of enemies., §7, §7Using the proper §bRune §7when casting spells, §7from §bRunic Items §7grants 1 stack of §6Arcane, §6Vision Ѫ§7., §7, §7Each §6Arcane Vision Ѫ §7stack grants you §c+2%, §c§7damage on your §bRunic Spells§7., §7, §7At §c10 §7stacks, the spells also explode on hit., §7, §7Lose 1 stack after §c4s §7of not gaining a stack., , §7§8This item can be reforged!, §6§lLEGENDARY LEGGINGS
+     */
+    private val armorStackTierBonus by RepoPattern.pattern(
+        "combat.armorstack.armor",
+        "§6Tiered Bonus: (?<type>.*) \\((?<amount>\\d)\\/4\\)"
     )
 
     @SubscribeEvent
     fun onActionBar(event: ActionBarUpdateEvent) {
         if (!isEnabled()) return
-        val stacks = armorStackPattern.findMatcher(event.actionBar) {
-            "§6§l" + group("stack")
-        }.orEmpty()
-        display = stacks
+        stackPattern.findMatcher(event.actionBar) {
+            updateStack(group("stack").toInt(), group("symbol"))
+        } ?: resetStack()
+
+        if (config.armorStackDisplay) event.changeActionBar(event.actionBar.replace(Regex("$stackPattern"), "").trim())
```

Comment:
instead of converting the pattern to regex here, add this function to `RegexUtils` and use it here

```kt
    fun String.replace(pattern: Pattern, replacement: String): String = pattern.matcher(this).replaceAll(replacement)
```